### PR TITLE
Add taints and tolerations

### PR DESCRIPTION
**Pull request by corindwyer**
### Description of the Change
* Add agent instance and instance group taints attribute and job tolerations attribute.
* Set K8s restart policy to `Never`.

### CODE REVIEW

File: titus-server-master/src/main/java/com/netflix/titus/master/scheduler/constraint/AgentManagementConstraint.java [hunk 1]
```diff
@@ -214,4 +210,37 @@ private Result evaluateAgentInstanceAttributes(AgentInstance agentInstance) {
...
+    private boolean taskCanTolerateTaints(V3QueueableTask taskRequest, AgentInstanceGroup instanceGroup, AgentInstance instance) {
+        Set<String> taints = getTaints(instanceGroup, instance);
+        Set<String> tolerations = getTolerations(taskRequest);
+        if (taints.isEmpty() && tolerations.isEmpty()) {
+            return true;
+        }
+
+        for (String taint : taints) {
+            if (tolerations.contains(taint)) {
+                return true;
```
Comment (tbak) [inline titus-server-master/src/main/java/com/netflix/titus/master/scheduler/constraint/AgentManagementConstraint.java]: We should tolerate all taints.

Comment (corindwyer) [reply titus-server-master/src/main/java/com/netflix/titus/master/scheduler/constraint/AgentManagementConstraint.java]: This was a mistake. Fixed

File: titus-server-master/src/main/java/com/netflix/titus/master/scheduler/constraint/AgentManagementConstraint.java [hunk 2]
```diff
@@ -219,12 +219,12 @@ private boolean taskCanTolerateTaints(V3QueueableTask taskRequest, AgentInstance
         }
 
         for (String taint : taints) {
-            if (tolerations.contains(taint)) {
-                return true;
+            if (!tolerations.contains(taint)) {
```
Comment (tbak) [inline titus-server-master/src/main/java/com/netflix/titus/master/scheduler/constraint/AgentManagementConstraint.java]: Simplify with containsAll?

Comment (corindwyer) [reply titus-server-master/src/main/java/com/netflix/titus/master/scheduler/constraint/AgentManagementConstraint.java]: updated

**Pull request merged**